pkg/registry/apis/provisioning/jobs/legacy/commit_helper.go

```diff
@@ -0,0 +1,124 @@
+package legacy
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
```

Comment:
Same as we did for github, we probably should move everything `git` related to a `git` folder. That would keep the `legacy` and `git` code separate which will use or reuse for a multi-tenant world :)